### ISSUE TITLE
Option[SomeTrait] not supported?

### ISSUE THREAD

**Issue by caturner81** [FasterXML/jackson-module-scala#279]
``` scala
trait SomeTrait

case class Child(str: String) extends SomeTrait

case class Parent(child: Option[SomeTrait])

val mapper = new ObjectMapper()
mapper.registerModule(DefaultScalaModule)

val child = Child("something")
val parent = Parent(Some(child))

println(mapper.writeValueAsString(parent))
```

I expect the above to produce

``` json
{"child":{"str":"something"}}
```

Instead if produces

``` json
{"child":{}}
```

Is this expected for some reason?

**Comment by brharrington** [comment]
I think this is the same as #240.

**Comment by nbauernfeind** [comment]
I think you might be right @brharrington. What version of scala, jackson and jackson-module-scala are you using @caturner81?

**Comment by caturner81** [comment]
Using Scala 2.11.  I had tried both 2.7.5 and 2.8.0-rc2 which were all that was available at the time.  2.8.2 seems to work as expected, so looks like it got fixed somewhere in there.

**Comment by viirya** [comment]
Looks like the fixing #271 was merged after 2.8.0.rc2. Please check out

https://github.com/FasterXML/jackson-module-scala/compare/jackson-module-scala-2.8.0.rc2...master

**Comment by cowtowncoder** [comment]
Assuming fix was legit -- please re-open if not, along with version information of a more recent version that can reproduce problem (like 2.9.9)
